### PR TITLE
Scroll optimization

### PR DESCRIPTION
**Pull request by OlegGordiichuk**
1) Optimization of the hard balance calculations. (Now in background thread)
2) Remove of the old code that was just wasting of the memory.
3) Optimization of the tickers usage. (We should firstly select all tickers for specific wallet and than sort them. In this case we select tickers in O(1) was O(N) complexity) 
4) Reduce usage of the main thread .

### CODE REVIEW

File: Trust/Tokens/Storage/TokensDataStore.swift [hunk 1]
```diff
@@ -72,11 +75,18 @@ class TokensDataStore {
         add(tokens: [newToken])
     }
 
+    func preparedTickres() -> [CoinTicker] {
```
Comment (vikmeup) [inline Trust/Tokens/Storage/TokensDataStore.swift]: `preparedTickres ` => `preparedTickers `

File: Trust/Tokens/ViewControllers/TokensViewController.swift [hunk 2]
```diff
@@ -49,13 +48,16 @@ final class TokensViewController: UIViewController {
         return footerView
     }()
 
+    let tableView: UITableView = {
```
Comment (vikmeup) [inline Trust/Tokens/ViewControllers/TokensViewController.swift]: if you initially it here, can you move all the configuration into this block?

File: Trust/Tokens/ViewControllers/TokensViewController.swift [hunk 3]
```diff
@@ -158,8 +153,9 @@ final class TokensViewController: UIViewController {
     }
 
     @objc func didBecomeActive() {
-        sheduleBalanceUpdate()
         startTokenObservation()
+        fetch()
+        sheduleBalanceUpdate()
```
Comment (vikmeup) [inline Trust/Tokens/ViewControllers/TokensViewController.swift]: `sheduleBalanceUpdate ` => `scheduleBalanceUpdate `

File: Trust/Transfer/ViewModels/SendViewModel.swift [hunk 4]
```diff
@@ -154,14 +154,14 @@ struct SendViewModel {
     }
     /// Get pair price with ticker
     func currentPairPrice() -> Decimal? {
-        guard let currentTokenInfo = storage.tickers().first(where: { $0.contract == destinationAddress.description }), let price = Decimal(string: currentTokenInfo.price) else {
+        guard let currentTokenInfo = storage.tickers.first(where: { $0.contract == destinationAddress.description }), let price = Decimal(string: currentTokenInfo.price) else {
```
Comment (vikmeup) [inline Trust/Transfer/ViewModels/SendViewModel.swift]: can you move this into one function? this this logic reused below as well

File: Trust/Tokens/Storage/TokensDataStore.swift (hunk 1 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/Storage/TokensDataStore.swift]: done

File: Trust/Tokens/ViewControllers/TokensViewController.swift (hunk 2 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewControllers/TokensViewController.swift]: yeah totally

File: Trust/Tokens/ViewControllers/TokensViewController.swift (hunk 3 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewControllers/TokensViewController.swift]: done

File: Trust/Transfer/ViewModels/SendViewModel.swift (hunk 4 above)
Comment (OlegGordiichuk) [reply Trust/Transfer/ViewModels/SendViewModel.swift]: done

File: Trust/Tokens/ViewControllers/TokensViewController.swift (hunk 2 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewControllers/TokensViewController.swift]: done

**Pull request closed without merging**